### PR TITLE
fix: Use column title as alias in bulk insert return statement

### LINKED ISSUE #7945: 🐛 Bug: Insert data API(v2) not returning proper response
### Please confirm if bug report does NOT exist already ?

- [X] I confirm there is no existing issue for this

### Steps to reproduce ?

Ref: https://discord.com/channels/661905455894888490/661905455894888493/1220838010715897986

### Desired Behavior

Should return record with column alias and not with actual column name.

### Project Details

Since 0.204.1

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

- Closes  #7945

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

## Summary by CodeRabbit

- **Refactor**
	- Enhanced data handling by transitioning from array to key-value pair storage for improved efficiency.
	- Expanded support in the `includes` method to accommodate more UI types for a broader application scope.

### CODE REVIEW

File: packages/nocodb/src/db/BaseModelSqlv2.ts [hunk 1]
```diff
@@ -3582,19 +3582,19 @@ class BaseModelSqlv2 {
...
+            returningObj[col.title] = col.column_name;
           }
         }
 
         responses =
           !raw && (this.isPg || this.isMssql)
             ? await trx
                 .batchInsert(this.tnPath, insertDatas, chunkSize)
-                .returning(this.model.primaryKeys?.length ? returningArr : '*')
+                .returning(this.model.primaryKeys?.length ? returningObj : '*')
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nocodb/src/db/BaseModelSqlv2.ts:3597]: We want `title` in response and not `column_name`

The query we are trying to populate is `"column_name" as "title"`

File: packages/nocodb/src/db/BaseModelSqlv2.ts [hunk 2]
```diff
@@ -3582,19 +3582,19 @@ class BaseModelSqlv2 {
           );
         }
       } else {
-        const returningArr: string[] = [];
+        const returningObj: Record<string, any> = {};
 
         if (!raw) {
           for (const col of this.model.primaryKeys) {
-            returningArr.push(col.column_name);
+            returningObj[col.title] = col.column_name;
```
Comment (mertmit) [inline packages/nocodb/src/db/BaseModelSqlv2.ts:3589]: Is it possible for us to use Record<string, string> and add a check on loop if required, bc providing undefined or null can break the query

Comment (pranavxc) [reply packages/nocodb/src/db/BaseModelSqlv2.ts:3589]: oh yes, will do that

Comment (pranavxc) [reply packages/nocodb/src/db/BaseModelSqlv2.ts:3589]: Actually primary key column_name and title will never be undefinec/null

**Review by mertmit — APPROVED** [review]

**Pull request merged**